### PR TITLE
OH2-305 | Suppliers create and update

### PR DESCRIPTION
**Pull request by SilverD3**
The API doesn't provide any endpoint for suppliers deletion, so the delete feature isn't include in this PR.

See OH2-305

### CODE REVIEW

**Comment by mwithi** [comment]
> The API doesn't provide any endpoint for suppliers deletion, so the delete feature isn't include in this PR.

for suppliers we have soft-deletion (updateSupplier with SUP_DELETED='Y') I don't know why supDeleted has not been mapped in the DTO, could you open an issue for that in OP project?

**Comment by SilverD3** [comment]
@mwithi There are too many boards in the OP project. In which one should open the issue ?

**Comment by mwithi** [comment]
> @mwithi There are too many boards in the OP project. In which one should open the issue ?

this one now has conflict, please I've no time to review all of them, let me ask @SteveGT96

File: src/components/accessories/admin/suppliers/editSupplier/EditSupplier.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  );
+  const navigate = useNavigate();
+
+  const handleSubmit = (value: SupplierDTO) => {
+    dispatch(updateSupplier(value));
+  };
+
+  useEffect(() => {
+    if (state?.supId !== Number(id)) {
+      navigate(PATHS.suppliers);
```
Comment (SteveGT96) [inline src/components/accessories/admin/suppliers/editSupplier/EditSupplier.tsx]:
```suggestion
      navigate(PATHS.admin_suppliers);
```

File: src/components/accessories/admin/suppliers/Suppliers.tsx [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+    });
+  };
+
+  return (
+    <SuppliersTable
+      onEdit={handleEdit}
+      headerActions={
+        <Button
+          onClick={() => {
+            navigate("./new");
```
Comment (SteveGT96) [inline src/components/accessories/admin/suppliers/Suppliers.tsx]:
```suggestion
            navigate(PATHS.admin_suppliers_new);
```

File: src/consts.ts [hunk 3]
```diff
@@ -24,6 +24,11 @@ export const PATHS = {
   diseases: "/admin/diseases",
   diseases_new: "/admin/diseases/new",
   diseases_edit: "/admin/diseases/#id/edit",
+  operations: "/admin/operations",
+  operations_edit: "/admin/operations/#id/edit",
+  suppliers: "/admin/suppliers",
+  suppliers_new: "/admin/suppliers/new",
+  suppliers_edit: "/admin/suppliers/:id/edit",
```
Comment (SteveGT96) [inline src/consts.ts]:
```suggestion
  admin_suppliers: "/admin/suppliers",
  admin_suppliers_new: "/admin/suppliers/new",
  admin_suppliers_edit: "/admin/suppliers/:id/edit",
```

**Review by SteveGT96 — CHANGES_REQUESTED** [review]
Now we are prefixing admin paths with "admin_" and we are reusing defined paths anywhere wanna navigate(navigate(PATHS....)

Please, could you align also ? Could check othe PRs to see.

**Review by SteveGT96 — APPROVED** [review]

**Review by mwithi — APPROVED** [review]
Approved by @SteveGT96

**Comment by mwithi** [comment]
Do we need new permissions for the new page?

image

**Comment by gasp** [comment]
Good catch @mwithi , issuing a fix immediately: https://github.com/informatici/openhospital-ui/pull/600

**Comment by mwithi** [comment]
> @mwithi There are too many boards in the OP project. In which one should open the issue ?

The board is the [OH2 Kanban](https://openhospital.atlassian.net/jira/software/c/projects/OH2/boards/12) but the issue has already been opened and backend side done, see OH2-310.

**Pull request merged**